flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedClientHandler.java

```diff
@@ -272,4 +316,53 @@ private void decodeBufferOrEvent(RemoteInputChannel inputChannel, NettyMessage.B
 			bufferOrEvent.releaseBuffer();
 		}
 	}
+
+	private void writeAndFlushNextMessageIfPossible(Channel channel) {
+		if (channelError.get() != null) {
+			return;
+		}
+
+		if (channel.isWritable()) {
```

Comment:
I guess you suggest to separate this PR into some smaller ones?

Reply:
Sorry, I could be more clear :) I meant that shorter of the if/else branch should go first for readability. Here instead of:
```
if (channel.isWritable()) {
    // long
    // block
    {
        // of
        {
            // nested
            // code
        }
    }
}
```
you could have:
```
if (!channel.isWritable()) {
    return;
}
// long
// block
{
    // of
    {
        // nested
        // code
    }
}
}
```
which is easier to understand because reader immediately knows that one of the branch is super simple and code doesn't nest that deeply.

Reply:
I am clear now and agree with that.
I am also used to the smaller branch before, but this part is referred to the current similar logic in `PartitionRequestQueue`.
I will fix that in the following modifications. :)